Modules/System/AL Rest Client/src/ALHttpRequestMessage.Codeunit.al

```diff
@@ -0,0 +1,87 @@
+    /// <param name="Method">The HTTP method to use. Valid options are GET, POST, PATCH, PUT, DELETE, HEAD, OPTIONS</param>
+    /// <remarks>Default method is GET</remarks>
+    procedure SetHttpMethod(Method: Text)
+    begin
+        AssertInitialized();
+        HttpRequestMessage.Method := Method;
+    end;
+
+    /// <summary>Sets the HTTP method for the HttpRequestMessage object.</summary>
+    /// <param name="Method">The HTTP method to use.</param>
+    /// <remarks>Default method is GET</remarks>
+    procedure SetHttpMethod(Method: Enum "Http Method")
+    begin
+        SetHttpMethod(Method.Names.Get(Method.Ordinals.IndexOf(Method.AsInteger())));
+    end;
+
+    /// <summary>Sets the Uri used for the HttpRequestMessage object.</summary>
+    /// <param name="Uri">The Uri to use for the HTTP request.</param>
+    /// <remarks>The valued must not be a relative URI.</remarks>
+    procedure SetRequestUri(Uri: Text)
+    begin
+        AssertInitialized();
+        HttpRequestMessage.SetRequestUri(Uri);
+    end;
+
+    /// <summary>Adds a request header to the HttpRequestMessage object.</summary>
+    /// <param name="HeaderName">The name of the header to add.</param>
+    /// <param name="HeaderValue">The value of the header to add.</param>
+    /// <remarks>If the header already exists, it will be overwritten.</remarks>
+    [NonDebuggable]
+    procedure AddRequestHeader(HeaderName: Text; HeaderValue: Text)
+    var
+        RequestHeaders: HttpHeaders;
+    begin
+        AssertInitialized();
+        HttpRequestMessage.GetHeaders(RequestHeaders);
+        if RequestHeaders.Contains(HeaderName) then
+            RequestHeaders.Remove(HeaderName);
+        RequestHeaders.Add(HeaderName, HeaderValue);
+    end;
+
+    /// <summary>Sets the HttpRequestMessage that is represented by the AL HttpRequestMessage object.</summary>
+    /// <param name="RequestMessage">The HttpRequestMessage to set.</param>
+    procedure SetHttpRequestMessage(var RequestMessage: HttpRequestMessage)
+    begin
+        AssertInitialized();
+        HttpRequestMessage := RequestMessage;
+    end;
+
+    /// <summary>Sets the content of the HttpRequestMessage that is represented by the AL HttpRequestMessage object.</summary>
+    /// <param name="ALHttpContent">The AL Http Content object to set.</param>
+    procedure SetContent(ALHttpContent: Codeunit "AL Http Content")
+    begin
+        AssertInitialized();
+        HttpRequestMessage.Content := ALHttpContent.GetHttpContent();
+    end;
+
+    /// <summary>Gets the HttpRequestMessage that is represented by the AL HttpRequestMessage object.</summary>
+    /// <returns>The HttpRequestMessage that is represented by the AL HttpRequestMessage object.</returns>
+    procedure GetRequestMessage() ReturnValue: HttpRequestMessage
+    begin
+        AssertInitialized();
+        ReturnValue := HttpRequestMessage;
+    end;
+
+    local procedure AssertInitialized()
```

Comment:
I don't understand the intention with using this procedure. Essentially, it makes sure that the consumer will always have to call `Initialize` before using any of the functions, but the actual code of `Initialize` just clears the global variable.

From the procedures on this codeunit it should be quite clear to any developer that it holds state, and that if they want to work with a new request message, they should just call `Clear(ALHttpRequestMessage)`, which would clear the global variables.

Reply:
The same applies to other codeunits that use the same pattern.

Reply:
Removed the Initialized pattern on the "Request Message" and "Response Message" codeunits.
But kept it on the "Rest Client" codeunit because it needs to be initialized with default values.